observer/probers/tls/tls.go

```diff
@@ -0,0 +1,167 @@
+package probers
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/crypto/ocsp"
+	"io"
+	"net/http"
+	"time"
+)
+
+const (
+	Success = iota
+	UnexpectedRoot
```

Comment:
Starting here we'll setup a new enum (`iota`):

```suggestion
	unexpectedRoot badOutcome = iota
```

Reply:
Next you can write an error type:

```go
type badOutcomeError struct {
	reason reason
}

func (e badOutcomeError) Error() string {
	switch e.Reason {
	case unexpectedRoot:
		return "..."
	case nextReason:
		return "..."
	case yetAnotherReason:
		return "..."
    ... and so on.
	}
	return "unknown error"
}
```

Reply:
All of that being said you could take the simpler tack of using var declarations of errors at the top of the package as well:

```go
var (
    unexpectedRootErr = errors.New("unexpected root")
    ...
)
```

If you return this error from one of the validation funcs to `Probe()`, when you call `err.Error()` you would get a string you could put in the label but the only way you could be absolutely sure that you weren't about to dump the contents of some random error into a label would be to check for each type of error you declare here.